src/server/routing/traits.rs

```diff
@@ -181,14 +181,6 @@ impl SyncNotificationHandler for DidChangeWatchedFiles {
         requester: &mut Requester<'_>,
         params: DidChangeWatchedFilesParams,
     ) -> LSPResult<()> {
-        // Invalidate changed cairo files.
-        for change in &params.changes {
-            if is_cairo_file_path(&change.uri) {
-                let Some(file) = state.db.file_for_url(&change.uri) else { continue };
-                PrivRawFileContentQuery.in_db_mut(state.db.as_files_group_mut()).invalidate(&file);
```

Comment:
just to make sure: have you verified that this works?

Reply:
It works fine for me, but if you see any case where this can break, please point it here

Reply:
It is weird because these invalidations are required according to https://murek.dev/salsa-book-0.16/common_patterns/on_demand_inputs.html

Reply:
How have you tested this?

Reply:
We do synthetic read in `priv_raw_file_content`, no? So if the file content has changed it will be invalidated